tests/test_vcs.py

```diff
@@ -143,3 +50,152 @@ def unknown_repo_type_url(request):
 def test_identify_raise_on_unknown_repo(unknown_repo_type_url):
     with pytest.raises(exceptions.UnknownRepoType):
         vcs.identify_repo(unknown_repo_type_url)
+
+
+def test_prompt_should_ask_and_rm_repo_dir(mocker, tmpdir):
+    mock_read_user = mocker.patch(
+        'cookiecutter.vcs.read_user_yes_no',
+        return_value=True,
+        autospec=True
+    )
+    repo_dir = tmpdir.mkdir('repo')
+
+    vcs.prompt_and_delete_repo(str(repo_dir))
+
+    assert mock_read_user.called
+    assert not repo_dir.exists()
+
+
+def test_prompt_should_ask_and_keep_repo_dir(mocker, tmpdir):
+    mock_read_user = mocker.patch(
+        'cookiecutter.vcs.read_user_yes_no',
+        return_value=False,
+        autospec=True
+    )
+    repo_dir = tmpdir.mkdir('repo')
+
+    with pytest.raises(SystemExit):
+        vcs.prompt_and_delete_repo(str(repo_dir))
+
+    assert mock_read_user.called
+    assert repo_dir.exists()
+
+
+def test_prompt_should_not_ask_if_no_input_and_rm_repo_dir(mocker, tmpdir):
+    mock_read_user = mocker.patch(
+        'cookiecutter.vcs.read_user_yes_no',
```

Comment:
Seems like it would be clearer to patch `cookiecutter.prompt.read_user_yes_no`, rather than the imported version of the function. (Unless this is following some pytest/patch best practice that I don't know about?)